### PR TITLE
feature/update-time-service: add time difference to time service

### PR DESCRIPTION
**Pull request by t-sats**

### CODE REVIEW

File: src/time_service/index.ts [hunk 1]
```diff
@@ -2,12 +2,30 @@ import fetch from 'node-fetch';
...
 // TODO calc delivery time for request
-export class TimeServiceClient implements TimeService {
-    async getTime(): Promise<number> {
+class TimeServiceClient implements TimeService {
+
+    private timeDifference: number;
+
+    constructor() {
+        this.timeDifference = 0;
+        this.init();
```
Comment (macaqueee) [inline src/time_service/index.ts:17]: you are calling async function here. so constructor will return Promise, not instance.

File: src/time_service/index.ts [hunk 2]
```diff
@@ -2,12 +2,30 @@ import fetch from 'node-fetch';
 import { config } from '../time_service/config';
 
 interface TimeService {
-    getTime(date: Date): Promise<number>;
+    getTimeFromServer(date: Date): Promise<number>;
```
Comment (macaqueee) [inline src/time_service/index.ts]: is it correct signature?

**Review by macaqueee — APPROVED** [review]

File: src/time_service/index.ts [hunk 3]
```diff
@@ -1,14 +1,34 @@
...
+    private timeDifference: number;
+
+    constructor( config?: Config ) {
+        this.config = config || new Config();
+        this.timeDifference = 0;
+        this.init();
+    }
+
+    private init(): void {
+        this.getTimeFromServer().then((networkTime: number) => {
```
Comment (Verbalist) [inline src/time_service/index.ts:21]: перепиши в async стиле

File: src/time_service/index.ts [hunk 4]
```diff
@@ -1,14 +1,34 @@
 import fetch from 'node-fetch';
-import { config } from '../time_service/config';
+import { Config } from '../time_service/config';
 
 interface TimeService {
-    getTime(date: Date): Promise<number>;
+    getTime();
```
Comment (Verbalist) [inline src/time_service/index.ts]: : number

**Review by Verbalist — CHANGES_REQUESTED** [review]

File: src/time_service/config.ts [hunk 5]
```diff
@@ -1,3 +1,17 @@
...
-};
+export class Config {
+    PATH: string;
+    PROTOCOL: string;
+    HOST: string;
+    PORT: string;
+
+    constructor() {
+        this.PATH = process.env.TIME_SERVER_HOST || 'getTime';
+        this.PROTOCOL = process.env.protocol || 'http';
```
Comment (macaqueee) [inline src/time_service/config.ts]: add `TIME_SERVER` prefix to properties in `process.env`

File: src/time_service/config.ts [hunk 6]
```diff
@@ -1,3 +1,17 @@
...
+        this.HOST = process.env.host || '0.0.0.0';
```
Comment (macaqueee) [inline src/time_service/config.ts]: lets rename to `TIME_SERVER_HOST`

**Review by Verbalist — APPROVED** [review]

**Pull request merged**